benchmark/README.md

```diff
@@ -0,0 +1,16 @@
+Benchmarking
+============
+
+This sub-project contains micro-benchmarks for commonly used GAX functionalities.
+
+To run a benchmark, run the following command
+```
+# Run from project root directory
+./gradlew benchmark:run -PcaliperArgs="['com.google.api.gax.grpc.CallableBenchmark']"
+```
+substituting any benchmark class for `com.google.api.gax.grpc.CallableBenchmark`.
```

Comment:
This is an interesting thought. I'm not sure how to do this well. Caliper accepts exactly one class to run in one invocation. You might also want to pass different options to different classes, maybe to set benchmark parameters. When we work to automate benchmarks, I can see us having a list of benchmarks and options to run them with. A command would just invoke Caliper a bunch of times. I'd like to do this a little later; I'm still experimenting with all the Caliper features.

The CLI provides an easy way to bootstrap until we can get these "curated benchmarks" in place. It's still valuable in the long run though: when we have 20 benchmarks, we'd probably want a way to only run one of them while optimizing.